app/assets/javascripts/components/util/routes.jsx

```diff
@@ -20,32 +22,48 @@ const TrainingApp = lazy(() => import('../../training/components/training_app.js
 const ActiveCoursesHandler = lazy(() => import('../active_courses/active_courses_handler.jsx'));
 const CoursesByWikiHandler = lazy(() => import('../courses_by_wiki/courses_by_wiki_handler.jsx'));
 
-const routes = (
-  <Suspense fallback={<Loading />}>
-    <Routes>
-      <Route path="/onboarding/*" element={<Onboarding />} />
-      <Route path="/recent-activity/*" element={<RecentActivityHandler />} />
-      <Route path="/courses/:course_school/:course_title/*" element={<Course />} />
-      <Route path="/course_creator" element={<ConnectedCourseCreator />} />
-      <Route path="/users/:username" element={<UserProfile />} />
-      <Route path="/alerts_list" element={<AdminAlerts />} />
-      <Route path="/settings" element={<SettingsHandler />} />
-      <Route path="/article_finder" element={<ArticleFinder />} />
-      <Route path="/training/*" element={<TrainingApp />} />
-      <Route path="/tickets/dashboard" element={<TicketsHandler />} />
-      <Route path="/tickets/dashboard/:id" element={<TicketShowHandler />} />
-      <Route path="/campaigns/*" element={<CampaignsHandler />} />
-      <Route path="/tagged_courses/:tag/alerts" element={<TaggedCourseAlerts />} />
-      <Route index element={<DetailedCampaignList headerText={I18n.t('campaign.campaigns')} userOnly={true}/>} />
-      <Route path="/dashboard" element={<DetailedCampaignList headerText={I18n.t('campaign.campaigns')} userOnly={true}/>} />
-      <Route path="/explore" element={<Explore dashboardTitle={window.dashboardTitle}/>} />
-      <Route path="/active_courses" element={<ActiveCoursesHandler dashboardTitle={window.dashboardTitle}/>}/>
-      <Route path="/courses_by_wiki/:wiki_url" element={<CoursesByWikiHandler />}/>
+const routes = () => {
+  const location = useLocation();
+  const dispatch = useDispatch();
+  const refreshArgs = {
+    lastUserRequestTimestamp: useSelector(state => state.users.lastRequestTimestamp),
+    courseSlug: useSelector(state => state.course.slug),
+    limit: useSelector(state => state.articles.limit),
```

Comment:
I know `articles` is the only thing we use limit for, but I think there's still potential for confusion later on, with this called `limit` instead of `articleLimit`. Now that this bundle of args is shared across several different types of requests, a more explicit name would help.